python/master.py

```diff
@@ -0,0 +1,321 @@
+    
+    The format for inputting matrix should have columns correspond to the celllines.
+    
+    """
+    Rmat = (len(gene_list), len(histone_list))
+    Rmat = np.zeros(Rmat)
+    Pmat = Rmat
+    i = 0 
+    for gene in gene_list:
+        j = 0
+        for histone in histone_list:
+            data1.loc[gene] = data1.loc[gene].fillna(method='ffill')
+            data2.loc[histone] = data2.loc[histone].fillna(method='ffill')
+            correlation = pearsonr(data1.loc[gene], data2.loc[histone])
+            Rmat[i][j] = correlation[0]
+            Pmat[i][j]
+            j = j+1
+        i = i+1
+    return Rmat, Pmat
+    
+def pearson_dfs(gene_list, histone_list, data1, data2):
+    """This functions purpose is to create two dataframes. The first dataframe will contain all the r-values between the two sets of data. The second will contain all the p-values for each r-value.
+    
+    This function is very similar to the Matrix function however it will create a dataframe with columns being histones and rows being genes. 
+   
+    This function is very specific. The dataframe created will contain histone markers as the columns and gene names as the index. 
+    
+    This function assumes you already took the intersection between the two arrays. If you do not do that, then the matrix dimensions will not agree, resulting in an error.
+    
+    The format for inputting matrix should have columns correspond to the celllines.
+    """
+    Rmat = (len(gene_list), len(histone_list))
+    Rmat = np.zeros(Rmat)
+    Pmat = Rmat[:]
+    i = 0 
+    for gene in gene_list:
+        j = 0
+        for histone in histone_list:
+            data1.loc[gene] = data1.loc[gene].fillna(method='ffill')
+            data2.loc[histone] = data2.loc[histone].fillna(method='ffill')
+            correlation = pearsonr(data1.loc[gene], data2.loc[histone])
+            Rmat[i][j] = correlation[0]
+            Pmat[i][j]
+            j = j+1
+        i = i+1
+        
+    Rdf = pd.DataFrame(Rmat)
+    Rdf.columns = histone_list
+    Rdf.insert(0, 'Genes', gene_list, True)
+    Rdf = Rdf.set_index('Genes')
+    
+    Pdf = pd.DataFrame(Pmat)
+    Pdf.columns = histone_list
+    Pdf.insert(0, 'Genes', gene_list, True)
+    Pdf = Pdf.set_index('Genes')
+    return Rdf, Pdf
+
+
+def heatmap(data, ylabels, size, colour, title):
+    """Using dataframes based on the pearson_dfs fucntion, this function will create a simple seaborn heatmap. By using a dataframe the xlabels will be automatically created from the dataframes columns
+    """
+    plt.figure(figsize=size)
+    plt.xlabel('Histone Markers')
+    plt.title(title)
+    ax = sns.heatmap(data, cmap = colour, square = True, yticklabels = ylabels)
+    ax.set(xlabel = 'Histone Markers')
+    plt.show
+
+
+
+    
+def plotlyheat(df, size, title, xaxis, yaxis):
```

Comment:
Put these in the vis.py script and add your name as an author.

Reply:
done